docs/smart-contracts/advanced/debug-cbor.md

```diff
@@ -0,0 +1,412 @@
+
+- data: This is arbitrary data that is passed as a parameter to the script. Most times this data is what is actually called the “redeemer”, instead of the complete 4-uple.
+
+- ex_units: The budget for the script execution in memory and CPU units. These numbers are used to compute the fee and must be higher or equal to the actual units used by the script execution. Execution units are computed according to the cost model, part of the protocol parameters of the Cardano blockchain. There is also a limit of the total memory and CPU units that all redeemers of a transaction can use. Also defined in the protocol parameters, Alonzo started with limits of 10.000 million units for CPU, and 10 million units for memory. However it was soon noticed that the memory limit was too low and was later raised to 16 million units. Execution units and their limits are a big issue in Cardano, as they impose important restrictions to smart contracts and developers must pay special attention to on-chain code optimization.
+
+In our example, we have only one redeemer for spending the script UTxO that is the first input according to the lexicographic order:
+
+```rust
+5: [[0, 0, 121([]), [1302238, 360901332]]]
+```
+
+We use 121([]) as the redeemer data to indicate to the script that we are trying to perform an increment operation. If it was 122([]), it would be a decrement operation. The script will validate, among other things, that the datum is updated according to the operation. The execution budget is 1302238 memory units and 360901332 CPU units, and was obtained in the balancing stage by doing an off-chain run of the validation.
+
+## Trivia
+
+To end, I propose you to try to answer the following questions. Thinking about them can be an interesting way to improve your understanding of the Cardano specification.
+
+- Is it possible to successfully submit a transaction with no wallet inputs?
+- Why it is not possible to successfully submit a transaction without signatures?
+- Is it possible to successfully submit a transaction with an empty list of inputs?
+- Is it possible to successfully submit a transaction with an empty list of outputs?
+
+Answer True or False to the following assertions:
+
+- Every transaction needs a collateral.
+- Every transaction with script inputs needs a collateral.
+- No transaction with no script inputs needs a collateral.
+
+## Runtime CBOR Debugging with Aiken
+
+While the previous sections focused on analyzing transactions at the blockchain and understanding what is CBOR and how to interpret it, developers need to debug CBOR data during smart contract development. This section covers techniques for inspecting individual values and data structures as you build and test your contracts.
+
+:::info
+You can read more at [Aiken - CBOR diagostic](https://aiken-lang.org/language-tour/troubleshooting#cbor-diagnostic) section.
```

Comment:
```suggestion
:::info
You can read more at [Aiken - CBOR diagostic](https://aiken-lang.org/language-tour/troubleshooting#cbor-diagnostic) section.
:::
```
The missing closure here causes the `(!) INFO` admonition to run all the way to the bottom of the page.